public/js/extra.js

```diff
@@ -179,6 +179,7 @@ function slugifyWithUTF8 (text) {
 
 export function isValidURL (str) {
   const pattern = new RegExp('^(https?:\\/\\/)?' + // protocol
+        '(?:\\S+(?::\\S*)?@)?' + // basic auth
```

Comment:
How about we simplify the whole function to a native browser interface (which is missing from IE11, but otherwise fine):

```
export function isValidURL (str) {
  try {
    new URL(str)
  } catch(e) {
    return false
  }
 
  return true
}
```

Because the regex you provide there will eat itself, probably ending up in yet another RegexDoS.

Reply:
Great idea, done!